src/components/toolbar.tsx

```diff
@@ -70,12 +70,19 @@ export class ToolbarComponent extends BaseComponent<IProps, IState> {
       this.handleDragNewTile(tool, e);
     };
     const renderToolButtons = (toolbarModel: IToolbarModel) => {
-      const { ui: { selectedTileIds } } = this.stores;
+      const { problem, ui: { selectedTileIds } } = this.stores;
+      const { document } = this.props;
       return toolbarModel.map(toolButton => {
+        const tilesOfTypeCount = document.content?.getTilesOfType(toolButton.id).length || 0;
+        const limitedTileTypes = problem.config?.settings?.limitedTileTypes as Record<string, number>;
```

Comment:
> that there could be a property named something like `maxCount`

or maybe `maxTiles` would be better, since `maxCount` could be ambiguous (count of what?).